frontend/index.html

```diff
@@ -0,0 +1,157 @@
+      width: 100vw;
+      height: 100vh;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      flex-direction: column;
+    }
+
+    i {
+      font-size: 5em;
+      font-size: 25vw;
+    }
+
+    p {
+      font-size: 5em;
+      font-size: 10vw;
+      margin: .3em;
+    }
+
+    #season {
+      margin-bottom: 2em;
+    }
+  </style>
+
+  <script>
+    const { isToday, closestTo, isEqual } = dateFns
+    class Text extends Marcel.Plugin {
+      constructor() {
+        super({
+          defaultProps: {
+            text: '',
+            textAlign: 'center',
+            stylesvar: {}
+          },
+        })
+
+        this.temperature = document.getElementById('content')
+        this.icon = document.getElementById('icon')
+        this.season = document.getElementById('season')
+      }
+
+      mapIcon = key =>
+        ({
+          8: 'wi-day-sunny',
+          2: 'wi-day-thunderstorm',
+          3: 'wi-day-rain',
+          5: 'wi-day-rain',
+          6: 'wi-day-snow',
+          7: 'wi-day-cloudy',
+          9: 'wi-day-cloudy'
+        }[key] || 'wi-day-sunny')
+
+      parse(date, data, checkDayFn) {
+        const closestDate = closestTo(date, data.list.map(e => e.dt_txt))
+        const closestData = data.list.find(e => isEqual(e.dt_txt, closestDate))
+
+        const [weather] = closestData.weather
+        let iconKey = `${weather.id}`.substr(0, 1)
+        if (iconKey === 8 && weather.id > 800) {
+          iconKey = 9
+        }
+
+        return {
+          icon: this.mapIcon(iconKey),
+          temperature: `${this.formatNumber(closestData.main.temp)}° C`,
+        }
+      }
+
+      formatNumber = new Intl.NumberFormat('fr-FR', {
+        maximumSignificantDigits: 2
+      }).format
+
+      async loadWeather() {
+        const response = await fetch(
+          `https://api.openweathermap.org/data/2.5/forecast?id=${this.props.city}&units=metric&appid=${this.props.apiKey}`
+        )
+        if (!response.ok) throw Error('Invalid server response', response)
+
+        const data = await response.json()
+        const today = new Date()
+
+        return {
+          city: data.city.name,
+          today: this.parse(today, data, isToday),
+        }
+      }
+
+      getSeason() {
+        const date = new Date()
+        const month = date.getMonth()
+        if (3 <= month <= 5) {
+            return 'le printemps'
+        }
+
+        if (6 <= month <= 8) {
+            return "l'été"
+        }
+
+        if (9 <= month <= 11) {
+            return "l'automne"
+        }
+
+        // Months 12, 01, 02
+        return "l'hiver"
+      }
+
+      async update() {
+        if (!this.props.city || !this.props.apiKey) return
+        const response = await this.loadWeather()
+        this.icon.className = `wi ${response.today.icon}`
+        this.temperature.innerText = response.today.temperature
+        this.season.innerText = `C'est ${this.getSeason()}`
```

Comment:
If you want, there is a `this.setState` function which works exactly as React one. It allow you to trigger a new call to the render function.

You can then refer to the `this.state` in your render function to update the DOM.